R/merge_sce_list.R

```diff
@@ -0,0 +1,138 @@
+#'  information, all final rowData slot column names will be appended with the
+#'  given SCE's name, as `<column_name>-<sce_name>`.
+#'
+#' @param sce_list A named list of SingleCellExperiment objects
+#' @param batch_column A character value giving the resulting colData column name
+#'  to differentiate originating SingleCellExperiment objects Default value is `batch`.
+#' @param retain_coldata_columns_main A vector of colData columns which should be retained
+#'  in the main experiment of the final merged SCE object.
+#' @param retain_coldata_columns_altexps A vector of colData columns which should be retained
+#'  in the altExps experiments of the final merged SCE object.
+#' @param retain_altexps A logical indicating whether any present altExp slots should
+#'   be retained in the merged SCE. Default is `TRUE`.
+#' @param preserve_rowdata_columns A vector of column names that appear in originating
+#'  SingleCellExperiment objects' rowData slots which should not be renamed with
+#'  the given SingleCellExperiment object name. These are generally columns which are
+#'  not specific to the given library's preparation or statistics. For example,
+#'  such an array might contain items like "Gene", "ensembl_ids", "gene_symbol", etc.
+#'
+#' @return A SingleCellExperiment object containing all SingleCellExperiment objects
+#'   present in the inputted list
+#' @export
+#'
+#' @import SingleCellExperiment
+merge_sce_list <- function(sce_list = list(),
+                           batch_column = "batch",
+                           retain_altexps = TRUE,
+                           retain_coldata_columns_main = c("sum",
+                                                           "detected",
+                                                           "total",
+                                                           "subsets_mito_sum",
+                                                           "subsets_mito_detected",
+                                                           "subsets_mito_percent",
+                                                           "miQC_pass"),
+                           retain_coldata_columns_altexps = c(),
+                           preserve_rowdata_columns = NULL) {
+  # Ensure `sce_list` is named  ----------------------
+  if (is.null(names(sce_list))) {
+    stop("Individual SingleCellExperiment objects in `sce_list` must be named.")
+  }
+  # Ensure `sce_list` has >=2 items --------------------------------------------
+  if (length(sce_list) < 2) {
+    stop("The `sce_list` must contain 2 or more SingleCellExperiment objects to merge.")
+  }
+
+  # Subset SCEs to shared features and ensure appropriate naming ------------------
+
+  # First, obtain intersection among all SCE objects
+  shared_features <- sce_list |>
+    purrr::map(rownames) |>
+    purrr::reduce(intersect)
+
+
+  process_columns <- function(sce, sce_name, retain_columns) {
```

Comment:
I'd pull this out to a separate function, maybe with a more descriptive name. What kind of processing are we doing here? It would still have a doxygen comments, just no  `#' @export` line, so it is available only inside the package (by default).